tests/virtual_environment/test_virtual_environment.py

```diff
@@ -43,11 +43,8 @@
 def venv_name():
     """Use a random venv name each time, at least partly to expose any
     hidden assumptions about the name of the venv directory
-
-    Force the name to have a space in it to expose possible fragilities that way
     """
-    hex = uuid.uuid1().hex
-    return hex[:2] + " " + hex[2:4]
```

Comment:
I had a quick look at this, and I think the issue is to do with the spaces on the virtualenv pip.

When trying to use the venv pip I get the following error in the terminal due to the spaces in the shebang:
```
$  ls -a /tmp/pytest-of-carlos/pytest-31/test_create_virtual_environmen0/a7\ 1d/bin/
.   activate      activate.fish     easy_install      pip   pip3.5  python3    python-config
..  activate.csh  activate_this.py  easy_install-3.5  pip3  python  python3.5  wheel


$ /tmp/pytest-of-carlos/pytest-31/test_create_virtual_environmen0/a7\ 1d/bin/pip --version
bash: /tmp/pytest-of-carlos/pytest-31/test_create_virtual_environmen0/a7 1d/bin/pip: "/tmp/pytest-of-carlos/pytest-31/test_create_virtual_environmen0/a7: bad interpreter: No such file or directory
```
`pip`  file content (trying to scape the space with `\ ` doesn't make a differnece):
```python
#!"/tmp/pytest-of-carlos/pytest-31/test_create_virtual_environmen0/a7 1d/bin/python3"

# -*- coding: utf-8 -*-
import re
import sys

from pip._internal.cli.main import main

if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
```

This works:

```
$ /tmp/pytest-of-carlos/pytest-31/test_create_virtual_environmen0/a7\ 1d/bin/python -m pip --version
pip 20.3.4 from /tmp/pytest-of-carlos/pytest-31/test_create_virtual_environmen0/a7 1d/lib/python3.5/site-packages/pip (python 3.5)
```